### DISCUSSION TITLE
Chart extension - register plugin

### DISCUSSION THREAD

**Discussion by kachna96** [Megabit/Blazorise#2960] [Q&A] [answered]
Hi all, I would like to ask about Charts - currently we have a simple doughnut chart and we are thinking about adding a text inside, similarly to this image
![image](https://user-images.githubusercontent.com/22668596/134483383-7f1c3299-a98d-44d2-be7f-731949247920.png)
I found a SO answer saying that it could be done in Chart.js with registering some custom plugins [https://stackoverflow.com/a/43940327/5825567](https://stackoverflow.com/a/43940327/5825567)
My question is if it is possible to do in Blazorise charts? I've looked into source code and haven't found any method which I would be able to use, but maybe I overlooked something ...
Thank you!

**Comment by stsrki** [comment, marked as answer]
I have quickly tested to see how it could work. You need to execute the JS script directly after you include `chart.js` in the `index.html` to register custom plugins.

```html
<script src="https://cdn.jsdelivr.net/npm/chart.js@2.8.0"></script>

<script>
        Chart.pluginService.register({
            beforeDraw: function (chart) {
                var width = chart.chart.width,
                    height = chart.chart.height,
                    ctx = chart.chart.ctx;
                ctx.restore();
                var fontSize = (height / 114).toFixed(2);
                ctx.font = fontSize + "em sans-serif";
                ctx.textBaseline = "middle";
                var text = chart.config.options.elements.center.text,
                    textX = Math.round((width - ctx.measureText(text).width) / 2),
                    textY = height / 2;
                ctx.fillText(text, textX, textY);
                ctx.save();
            }
        });
    </script>
```

And for the options, you can define them as anonymous object:

```cshtml
<Chart @ref="doughnutChart" Type="ChartType.Doughnut" TItem="double" OptionsObject="@chartOptions" />
@code {
    object chartOptions = new
    {
        Elements = new
        {
            Center = new
            {
                Text = "50%"
            }
        },
        CutoutPercentage = 75,
        Legend = new
        {
            Display = false
        }
    };
}
```

**Result**

![image](https://user-images.githubusercontent.com/900302/134485724-dc1de566-b18e-40ca-8e1c-ab5c21f8f61a.png)

**Reply by kachna96** [reply]
Works like a charm, thank you!
![image](https://user-images.githubusercontent.com/22668596/134499057-6273062e-75ea-4a11-a544-309b636f4841.png)
